aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -86,6 +86,15 @@
      * Start subscribing to model mutations.
      */
     synchronized void startSubscriptions() {
+        // let's ensure that we are not duplicating subscriptions here
+        if (ongoingOperationsDisposable.size() > 0) {
```

Comment:
This `ongoingOperationsDisposable` is also used to track the disposable created in the `startDrainingMutationBuffer` method (line 207). It's possible (although not likely) that the disposable created when initializing subscriptions is disposed, but the other one (from line 207) is not, so you'd still have 1 item in the `ongoingOperationsDisposable` variable.

In that case, subscriptions wouldn't be restarted. Maybe one thing we could do here is track those disposables separately rather than lumping them together. If we do that, we would also need to update the `stopAllSubscriptionActivity` method to dispose of both.